public/app/features/alerting/unified/reducers/alertmanager/notificationPolicyRoutes.test.ts

```diff
@@ -0,0 +1,132 @@
+  const A_identifier = rootRouteWithIdentifiers.routes![0].id;
+
+  const initialConfig: AlertManagerCortexConfig = {
+    alertmanager_config: {
+      time_intervals: [],
+      mute_time_intervals: [],
+      route: rootRouteWithIdentifiers,
+    },
+    template_files: {},
+  };
+  it('Should add a new route with receiver E as a child of default route', () => {
+    const routeWithID = addUniqueIdentifierToRoute({
+      receiver: 'E',
+    });
+    const newFormRoute: Partial<FormAmRoute> = {
+      receiver: 'E',
+      id: routeWithID.id,
+    };
+    const action = addRouteAction({
+      alertmanager: 'alertmanager',
+      partialRoute: newFormRoute,
+      referenceRouteIdentifier: ROOT_identifier,
+      insertPosition: 'child',
+    });
+
+    expect(routesReducer(initialConfig, action)).toMatchSnapshot();
+  });
+
+  it('Should not add a new route with receiver E as a child of route with receiver A, if data has been updated by another user', () => {
+    const routeWithID = addUniqueIdentifierToRoute({
+      receiver: 'E',
+    });
+    const newFormRoute: Partial<FormAmRoute> = {
+      receiver: 'E',
+      id: routeWithID.id,
+    };
+    const action = addRouteAction({
+      alertmanager: 'alertmanager',
+      partialRoute: newFormRoute,
+      referenceRouteIdentifier: A_identifier,
+      insertPosition: 'child',
+    });
+    const modifiedConfig = produce(initialConfig, (draft) => {
+      draft.alertmanager_config.route!.routes = [];
+    });
+    const rootRouteWithIdentifiers = addUniqueIdentifierToRoute(modifiedConfig.alertmanager_config.route!);
+    const modifiedConfigWithIdentifiers = produce(modifiedConfig, (draft) => {
+      draft.alertmanager_config.route = rootRouteWithIdentifiers;
+    });
+    expect(() => routesReducer(modifiedConfigWithIdentifiers, action)).toThrow();
+  });
+
+  it('Should update an existing route A with receiver B', () => {
+    const newRoute: Partial<FormAmRoute> = {
+      receiver: 'B',
+      id: A_identifier,
+    };
+    const action = updateRouteAction({
+      alertmanager: 'alertmanager',
+      update: newRoute,
+    });
+
+    expect(routesReducer(initialConfig, action)).toMatchSnapshot();
+  });
+
+  it('Should not update route if config has been updated by another user', () => {
+    const newRoute: Partial<FormAmRoute> = {
+      receiver: 'B',
+      id: A_identifier,
+    };
+    const action = updateRouteAction({
+      alertmanager: 'alertmanager',
+      update: newRoute,
+    });
+    const modifiedConfig = produce(initialConfig, (draft) => {
+      draft.alertmanager_config.route!.routes = [];
+    });
+    const rootRouteWithIdentifiers = addUniqueIdentifierToRoute(modifiedConfig.alertmanager_config.route!);
+    const modifiedConfigWithIdentifiers = produce(modifiedConfig, (draft) => {
+      draft.alertmanager_config.route = rootRouteWithIdentifiers;
+    });
+    expect(() => routesReducer(modifiedConfigWithIdentifiers, action)).toThrow();
```

Comment:
How does it know that the config has been updated by another user? I couldn't find any code responsible for comparing old and new configs in the `routesReducer`. Am I missing something?